pkg/apis/apps/v1alpha1/broadcastjob_types.go

```diff
@@ -123,8 +131,52 @@ type BroadcastJobStatus struct {
 	// The desired number of pods, this is typically equal to the number of nodes satisfied to run pods.
 	// +optional
 	Desired int32 `json:"desired" protobuf:"varint,7,opt,name=desired"`
+
+	// The phase of the job.
+	// +optional
+	Phase BroadcastJobPhase `json:"phase" protobuf:"varint,8,opt,name=phase"`
+}
+
+// BroadcastJobPhase indicates the phase of the job.
+type BroadcastJobPhase string
+
+const (
+	// PhaseCompleted means the job is completed.
+	PhaseCompleted BroadcastJobPhase = "completed"
+
+	// PhaseRunning means the job is running.
+	PhaseRunning BroadcastJobPhase = "running"
+
+	// PhasePaused means the job is paused.
+	PhasePaused BroadcastJobPhase = "paused"
+
+	// PhaseFailed means the job is failed.
+	PhaseFailed BroadcastJobPhase = "failed"
+)
+
+// FailurePolicy indicates the behavior of the job, when failed pod is found.
+type FailurePolicy struct {
+	// Type indicates the type of FailurePolicyType.
+	Type FailurePolicyType
+
+	// RestartLimit specifies the number of retries before marking the pod failed.
+	RestartLimit int32
```

Comment:
I think we should remove BackoffLimit *int32 from the API if we provide per pod restart limit. I don't see a case where people need both.